patches/minecraft/net/minecraft/item/Food.java.patch

```diff
@@ -0,0 +1,49 @@
+--- a/net/minecraft/item/Food.java
++++ b/net/minecraft/item/Food.java
+@@ -11,9 +11,9 @@
+    private final boolean field_221472_c;
+    private final boolean field_221473_d;
+    private final boolean field_221474_e;
+-   private final List<Pair<EffectInstance, Float>> field_221475_f;
++   private final List<Pair<java.util.function.Supplier<EffectInstance>, Float>> field_221475_f;
+ 
+-   private Food(int p_i50106_1_, float p_i50106_2_, boolean p_i50106_3_, boolean p_i50106_4_, boolean p_i50106_5_, List<Pair<EffectInstance, Float>> p_i50106_6_) {
++   private Food(int p_i50106_1_, float p_i50106_2_, boolean p_i50106_3_, boolean p_i50106_4_, boolean p_i50106_5_, List<Pair<java.util.function.Supplier<EffectInstance>, Float>> p_i50106_6_) {
```

Comment:
I have an AT on this constructor in one of my mods as I extend this class to replace hunger and saturation with suppliers (override the getters to pull from a server config file). Since ATs break when a Forge patches change the signature, could we change this to protected in the patch so I don't need my AT? Alternatively, if Forge does not want to change scope then can we keep the old constructor and have it call the new one?

I'm open for discussion on this comment, if anyone knows a better way to do what I'm doing without more patches. Adding a food interface I can implement instead would probably be better for my needs, but outside the scope of this PR.

Reply:
I am also pretty sure forge does not like overwriting constructors/built in methods. Instead they should be deprecated and create ones with suppliers. And then reference the new supplier ones from the deprecated ones. For example: https://github.com/MinecraftForge/MinecraftForge/blob/1.15.x/patches/minecraft/net/minecraft/block/FlowerPotBlock.java.patch

I believe the main reason forge does this is so it doesn't mess with the mapping process, though I could be wrong about that so if someone knows differently feel free to chime in.

Reply:
Yeah I was considering that although I thought they only did that in order to not break binary comparability therefore avoiding having to patch other vanilla classes however, this constructor is private so I assumed it was alright to change. But yes, if it’s to do with the mapping process or some other reason I shall update it accordingly.

Reply:
No it's nothing to do with binary compatibility. We don't remove vanilla methods.

Reply:
Alright, I'll add it back, although if I did, nothing would be calling it because it's only called by the build function in the food builder and I'm modifying that to pass a Supplier instead.